### PR TITLE
Bugfix/war 1031: Iterative testsuite overwrites datafile setting for next non-iterative suite

### PR DESCRIPTION
**Pull request by navvrnd**
Suite initialization not initializing suite fields of data_repository resulting in subsequent suites in the project using the previous suites values. Use pj_cli_part_1_def_other_attr from warrior_fnc_tests for testing this, which has been modified to test this.

### CODE REVIEW

**Comment by Annam16** [comment]
tested with "pj_cli_part_1_def_other_attr ". Working fine. hence approving this pull request

**Comment by Annam16** [comment]
@navvrnd Please resolve merge conflicts

File: warrior/WarriorCore/testsuite_driver.py [hunk 1]
```diff
@@ -51,13 +51,23 @@ def get_suite_details(testsuite_filepath, data_repository, from_project, res_sta
...
-    suite_name = Utils.xml_Utils.getChildTextbyParentTag(testsuite_filepath, 'Details', 'Name')
-    suite_title = Utils.xml_Utils.getChildTextbyParentTag(testsuite_filepath, 'Details', 'Title')
-    suite_exectype = testsuite_utils.get_exectype_from_xmlfile(testsuite_filepath)
-    def_on_error_action = Utils.testcase_Utils.get_defonerror_fromxml_file(testsuite_filepath)
-    def_on_error_value = Utils.xml_Utils.getChildAttributebyParentTag(testsuite_filepath,
-                                                                      'Details',
-                                                                      'default_onError', 'value')
+    print_info("initializing suite fields in data_repository")
+    suite_fields_to_initialize = ['wt_suite_execution_dir', 'suite_data_file',
+                                  'wt_suite_name', 'suite_exectype']
```
Comment (venkat1077) [inline warrior/WarriorCore/testsuite_driver.py]: Can you add inline comment for this initialization(why these 4 attribs)

Comment (navvrnd) [inline warrior/WarriorCore/testsuite_driver.py]: done.

**Comment by navvrnd** [comment]
addressed all review comments, please review and approve.

**Pull request merged**